library/core/src/intrinsics.rs

```diff
@@ -2063,18 +2276,22 @@ extern "rust-intrinsic" {
     /// takes the data pointer and a pointer to the target-specific exception
     /// object that was caught. For more information see the compiler's
     /// source as well as std's catch implementation.
+    #[rustc_nounwind]
     pub fn r#try(try_fn: fn(*mut u8), data: *mut u8, catch_fn: fn(*mut u8, *mut u8)) -> i32;
```

Comment:
Actually the `catch_fn` passed to `try` could unwind, which will unwind out of the `try` intrinsic. However this is a somewhat theoretical concern since our current implementation doesn't call any functions that unwind in the catch callback.

Reply:
This is an attribute on the signature, which correctly states that we cannot unwind out from the try intrinsic.

Reply:
If `catch_fn` unwinds, then the exception will also unwind out of the `try` intrinsic. Only unwinds from `try_fn` are caught.